apm-commons/apm-datacarrier/src/main/java/org/apache/skywalking/apm/commons/datacarrier/buffer/BufferItem.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.commons.datacarrier.buffer;
+
+/**
+ * Buffer item fills the whole the {@code org.apache.skywalking.apm.commons.datacarrier.buffer.Buffer}, and never be
+ * cleared from it. At the same time, it provide `volatile` field to hold the real data.
+ *
+ * @author wusheng
+ */
+public class BufferItem {
+    /**
+     * Hold the ref for the real data. `null` means no data.
+     */
+    private volatile Object item = null;
+
+    public Object getItem() {
+        return item;
+    }
+
+    public void setItem(Object item) {
+        this.item = item;
```

Comment:
Item set volatile may reduce the producer's performance

Reply:
Yes. The key is I want to reduce the visibility to consumer thread. I want to check whether it is real having that issue.

Reply:
The point is, if the visibility issue happens, it affect the performance more, the queue will be blocked longer than required.